src/main/java/uk/gov/hmcts/ccd/data/casedetails/query/CaseDetailsQueryBuilderFactory.java

```diff
@@ -3,24 +3,39 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.ccd.data.casedetails.CaseDetailsEntity;
+import uk.gov.hmcts.ccd.data.casedetails.search.MetaData;
 
 import javax.persistence.EntityManager;
+import java.util.List;
 
 @Component
 public class CaseDetailsQueryBuilderFactory {
 
     private final UserAuthorisationSecurity userAuthorisationSecurity;
+    private final List<CaseDetailsAuthorisationSecurity> caseDetailsAuthorisationSecurities;
```

Comment:
UserAuthorisationSecurity and CaseDetailsAuthorisationSecurity at the moment have slightly different apis (one returns void and accepts a metadata, the other accepts no metadata and returns the builder), but we could unify them and have both extend a common interface, e.g. `AuthorisationSecurity`. Or we could have UserAuthorisationSecurity be a CaseDetailsAuthorisationSecurity.

Reply:
@mario-paniccia , I originally made UserAuthorisationSecurity to implement CaseDetailsAuthorisationSecurity but then realised there is a method in DefaultCaseDetailsRepository that calls CaseDetailsQueryBuilderFactory.secure(em) without metadata. Its used for individual case clause. If I unify, I will have to pass metadata as null for this call and in the implementation for CaseStateAuthorisationSecurity and I will have to explicitly check if metadata is not null before adding the clause (which doesn't look great). 

It will be good to unify it, but couldn't find a common ground.